thriftrw/spec/struct.pyx

```diff
@@ -517,6 +519,9 @@ def struct_cls(struct_spec, scope):
         required_fields,
         list(zip(optional_fields, field_defaults)),
     )
+    if struct_spec.hashable:
+        struct_dct['__hash__'] = \
+            lambda self: hash(tuple([getattr(self, field.name) for field in struct_spec.fields]))
```

Comment:
This will be more readable if you use another function to generate the hash
function.

```
if struct_spec.hashable:
  struct_dct['__hash__'] = struct_hasher(struct_spec)
```


```
def struct_hasher(spec):
    def __hash__(self):
        return hash(tuple(getattr(self, f.name) for f in spec.fields))
    return __hash__
```